File: LeetCode/Python Solutions/First Unique Character in a String/first.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        # Count the frequency of each character
        frequency = {}
        for char in s:
            frequency[char] = frequency.get(char, 0) + 1
            
        # Find the first character that appears only once
        for i, char in enumerate(s):
            if frequency[char] == 1:
                return i
        return -1

File: LeetCode/Java Solutions/First Unique Character in a String/first.java
class Solution {
    public int firstUniqChar(String s) {
        // Store frequencies of each character
        int[] freq = new int[26]; // Assuming only lowercase English letters
        for (int i = 0; i < s.length(); i++) {
            freq[s.charAt(i) - 'a']++;
        }
        
        // Find the first non-repeating character
        for (int i = 0; i < s.length(); i++) {
            if (freq[s.charAt(i) - 'a'] == 1) {
                return i;
            }
        }
        return -1;
    }
}
